backend/api/views/book.py

```diff
@@ -19,6 +21,44 @@ def invalid_book_data(user_data):
         return True
 
 
+@book.route("/book_from_csv", methods=["POST"])
+def create_book_from_csv():
+    uploaded_csv = request.files["File"]
+    if not uploaded_csv:
+        return create_response(message="Missing CSV file", status=409)
+
+    stream = io.StringIO(uploaded_csv.stream.read().decode("UTF8"), newline=None)
+    parsed_data = csv.reader(stream)
+    header = next(parsed_data)
+    header = [col.replace(" ", "_") for col in next(parsed_data)]
+    for row in parsed_data:
+        row_dict = {
+            key.lower(): value for key, value in zip(header, row)
+        }  # resilient to future changes of column positions
+        if invalid_book_data(row_dict):
+            return create_response(
+                message=f"Invalid book data {row}",
+                status=409,
+                data={"status": "failure"},
```

Comment:
Nit: very minor but is there a reason to have data dict here, but not in the previous create_response: line 28 or 58